### PR TITLE
Invalid module location

### LINKED ISSUE #38110: Wrong suggestion in module declaration error message
Maybe this bug is related to #36146.
# Library structure
```
src/lib.rs
src/foo/mod.rs
src/foo/bar.rs
```

src/lib.rs:
`pub mod foo;`

src/foo/mod.rs:
`pub mod bar;`

src/foo/bar.rs
`pub mod baz;`

# Error
Currently:
```
error: cannot declare a new module at this location
 --> foo\bar.rs:1:9
  |
1 | pub mod baz;
  |         ^^^
  |
note: maybe move this module `foo` to its own directory via `foo/mod.rs`
 --> foo\bar.rs:1:9
  |
1 | pub mod baz;
  |         ^^^
```
Should be:
```
  error: cannot declare a new module at this location
   --> foo\bar.rs:1:9
    |
  1 | pub mod baz;
    |         ^^^
    |
- note: maybe move this module `foo` to its own directory via `foo/mod.rs`
+ note: maybe move this module `bar` to its own directory via `bar/mod.rs`
   --> foo\bar.rs:1:9
    |
  1 | pub mod baz;
    |         ^^^
```
Additionally, I don't understand what to make of the repeated code excerpt.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #38110.

r? @jseyfried

### CODE REVIEW

File: src/libsyntax/parse/parser.rs [hunk 1]
```diff
@@ -5363,26 +5365,30 @@ impl<'a> Parser<'a> {
...
-            } else {
-                Err(err)
             }
+            Err(err)
         } else {
-            paths.result.map_err(|err| self.span_fatal_err(id_sp, err))
+            match paths.result {
+                Ok(succ) => Ok(succ),
+                Err(err) => Err(self.span_fatal_err(id_sp, err)),
+            }
```
Comment (jseyfried) [inline src/libsyntax/parse/parser.rs]: Why this change? It looks good as is.

**Review by jseyfried — COMMENTED** [review]
Thanks! r=me modulo comment.

File: src/libsyntax/parse/parser.rs (hunk 1 above)
Comment (GuillaumeGomez) [inline src/libsyntax/parse/parser.rs]: A rebase error I guess. Removing it.

**Pull request merged**